ats/anomaly_detectors/base.py

```diff
@@ -34,3 +35,58 @@ def apply(self, data, *args, **kwargs):
         """
         raise NotImplementedError()
 
+class TimeseriaAnomalyDetector(AnomalyDetector):
+    """
+    Base class for anomaly detectors wrapped from the timeseria library.
+    """  
+    
+    model_class = None
+
+    def __init__(self, *args, **kwargs):
+        if self.model_class is None:
+            raise NotImplementedError('Subclasses must define a timeseria model')
+        self.model = self.model_class(*args, **kwargs)
+        self.apply_params = {}
+
+    def set_apply_params(self, **kwargs):
+        """
+        Set parameters for the apply method of the timeseria model.
+        """  
+        self.apply_params = kwargs
+    
+    def get_apply_params(self):
+        """
+        Get parameters for the apply method of the timeseria model.
+        """  
+        return self.apply_params
+
+    def fit(self, data, *args, **kwargs):
+        """
+        Fit the timeseria anomaly detector model.
+        """      
+        if not isinstance(data,pd.DataFrame):
+            raise NotImplementedError('Not yet implemented for non DataFrame inputs')
+        timeseries_df = data
+
+        # Using timeseria to fit the model
+        timeseries = convert_timeseries_df_to_timeseries(timeseries_df)
+        model = self.model
+        model.fit(timeseries, *args, **kwargs)
+
+    def apply(self, data):
+        """
+        Apply the timeseria anomaly detector model.
+        """     
+        if not isinstance(data,pd.DataFrame):
+            raise NotImplementedError('Not yet implemented for non DataFrame inputs')
+        timeseries_df = data
+
+        # Using timeseria to fit and apply the model
+        timeseries = convert_timeseries_df_to_timeseries(timeseries_df)
+        timeseries = self.model.apply(timeseries, **self.apply_params)
```

Comment:
Here you should use `get_apply_params()` for consistency. Otherwise, if one overwrites the `get_apply_params()` in a subclass for whatever reason, the apply would still use the internally saved params.